File: frontend_mobile/lib/data/models/user/cooking_dna_dto.dart
import 'package:json_annotation/json_annotation.dart';
import 'cuisine_stat_dto.dart';

part 'cooking_dna_dto.g.dart';

@JsonSerializable()
class CookingDnaDto {
  // XP & Level
  final int totalXp;
  final int level;
  final String levelName;
  final int xpForCurrentLevel;
  final int xpForNextLevel;
  final double levelProgress;

  // Cooking Stats
  final double successRate;
  final int totalLogs;
  final int successCount;
  final int partialCount;
  final int failedCount;

  // Streak
  final int currentStreak;
  final int longestStreak;

  // Cuisine Distribution
  final List<CuisineStatDto> cuisineDistribution;

  // Content counts
  final int recipeCount;
  final int logCount;
  final int savedCount;

  CookingDnaDto({
    required this.totalXp,
    required this.level,
    required this.levelName,
    required this.xpForCurrentLevel,
    required this.xpForNextLevel,
    required this.levelProgress,
    required this.successRate,
    required this.totalLogs,
    required this.successCount,
    required this.partialCount,
    required this.failedCount,
    required this.currentStreak,
    required this.longestStreak,
    required this.cuisineDistribution,
    required this.recipeCount,
    required this.logCount,
    required this.savedCount,
  });

  factory CookingDnaDto.fromJson(Map<String, dynamic> json) =>
      _$CookingDnaDtoFromJson(json);
  Map<String, dynamic> toJson() => _$CookingDnaDtoToJson(this);

  /// Get localized level name key for i18n
  String get levelNameKey => 'profile.$levelName';

  /// Get XP progress as percentage string
  String get xpProgressText => '$totalXp / $xpForNextLevel XP';

  /// Get success rate as percentage
  int get successRatePercent => (successRate * 100).round();
}

File: backend/src/main/java/com/pairingplanet/pairing_planet/dto/user/CookingDnaDto.java
package com.pairingplanet.pairing_planet.dto.user;

import lombok.Builder;

import java.util.List;

@Builder
public record CookingDnaDto(
        // XP & Level
        int totalXp,
        int level,
        String levelName,         // "beginner", "homeCook", "skilledCook", "homeChef", "expertChef", "masterChef"
        int xpForCurrentLevel,    // XP threshold for current level
        int xpForNextLevel,       // XP threshold for next level
        double levelProgress,     // Progress within current level (0.0 - 1.0)

        // Cooking Stats
        double successRate,       // Success rate (0.0 - 1.0)
        int totalLogs,            // Total number of cooking logs
        int successCount,         // Logs with SUCCESS outcome
        int partialCount,         // Logs with PARTIAL outcome
        int failedCount,          // Logs with FAILED outcome

        // Streak
        int currentStreak,        // Current consecutive cooking days
        int longestStreak,        // Longest streak ever achieved

        // Cuisine Distribution (top 5 + "other")
        List<CuisineStatDto> cuisineDistribution,

        // Content counts (same as profile)
        long recipeCount,
        long logCount,
        long savedCount
) {}
